shell/browser/electron_browser_context.cc

```diff
@@ -431,9 +436,32 @@ void ElectronBrowserContext::SetCorsOriginAccessListForOrigin(
     std::vector<network::mojom::CorsOriginPatternPtr> allow_patterns,
     std::vector<network::mojom::CorsOriginPatternPtr> block_patterns,
     base::OnceClosure closure) {
-  // TODO(nornagon): actually set the CORS access lists. This is called from
-  // extensions/browser/renderer_startup_helper.cc.
-  base::ThreadTaskRunnerHandle::Get()->PostTask(FROM_HERE, std::move(closure));
+  auto& context_map = ElectronBrowserContext::browser_context_map();
+  auto barrier_closure =
+      BarrierClosure(1 + context_map.size(), std::move(closure));
+
+  for (auto& iter : context_map) {
+    if (iter.second) {
```

Comment:
i'm not sure this can ever be `false`? `context_map` is `std::map<PartitionKey, std::unique_ptr<ElectronBrowserContext>>`, so `iter.second` is `std::unique_ptr`. I don't think it's ever null in our usage?

Reply:
A DCHECK might be a better way of signalling intent then; "this shouldn't ever be nullptr" is less ambiguous than "if we have a pointer"